x-pack/plugins/security_solution/public/resolver/store/middleware/resolver_tree_fetcher.ts

```diff
@@ -42,7 +43,7 @@ export function ResolverTreeFetcher(
   return async (id: string) => {
     // const id = 'alerts-page';
     const state = api.getState();
-    const databaseParameters = selectors.treeParametersToFetch(state.analyzer[id]);
+    let databaseParameters = selectors.treeParametersToFetch(state.analyzer[id]);
```

Comment:
I couldn't find a way to get the agent.id value into tree parameters before the dataAccessLayer.entity() is called. After it is called we update tree parameters with agentId value. Do you think it would still introduce a bug in this case?

Reply:
@kqualters-elastic Hey Kevin! Could you take another look here? I might be missing something, so looking up to your experience if we can get `agent.id` sooner. 
Thanks! :)

Reply:
when are the endpoint side changes being rolled out?

Reply:
@ferullo @brian-mckinney could you help with the endpoint release timeline?

Reply:
@tomsonpl I was mistaken, databaseDocumentId is not entity_id, but the _id of the doc, so what you had is how it should work. Sorry for the delays in response, but asking me to review these changes without having a way to run them myself due to 0 access to the data (as was the case with crowdstrike/sentinel one as well) is rough.

Reply:
Oh, sorry I though you mentioned the official release. To test it you can follow this instruction: https://github.com/elastic/security-team/issues/9340#issuecomment-2132436671

Reply:
Endpoint changes will go in 8.15.0 unless this PR isn't merged in time or any other issues are found in which case it can delay for 8.16.0.

This PR is backwards compatible right? So this is a question of deadline, not timing the two together? Endpoint team knows to wait for this PR before merging the change there.